Example Apps/Example ObjC/RemoteControlManager.m

```diff
@@ -0,0 +1,225 @@
+//
+//  RemoteControlManager.m
+//  SmartDeviceLink-Example-ObjC
+//
+//  Created by Beharry, Justin (J.S.) on 8/1/22.
+//  Copyright © 2022 smartdevicelink. All rights reserved.
+//
+
+#import "RemoteControlManager.h"
+#import "SmartDeviceLink.h"
+#import "AlertManager.h"
+
+@interface RemoteControlManager()
+
+@property (strong, nonatomic) SDLManager *sdlManager;
+@property (strong, nonatomic) SDLRemoteControlCapabilities *remoteControlCapabilities;
+@property (strong, nonatomic) NSString *climateModuleId;
+@property (strong, nonatomic) NSNumber<SDLBool> *consent;
+@property (strong, nonatomic) SDLClimateControlData* climateData;
+
+- (void)subscribeVehicleData;
+- (void)initializeClimateData;
+- (void)subscribeClimateControlData;
+- (void)turnOnAc;
+- (void)turnOffAc;
+- (void)setClimate;
+- (NSArray*)getButtons;
+
+@end
+
+@implementation RemoteControlManager
+
+- (instancetype)initWithManager:(SDLManager *)manager {
+    self = [super init];
+    if (!self) {
+        return nil;
+    }
+
+    _sdlManager = manager;
+    
+    [_sdlManager.systemCapabilityManager subscribeToCapabilityType:SDLSystemCapabilityTypeRemoteControl withUpdateHandler:^(SDLSystemCapability * _Nullable capability, BOOL subscribed, NSError * _Nullable error) {
+        if (!capability.remoteControlCapability) { return; }
+        
+        self->_remoteControlCapabilities = capability.remoteControlCapability;
+        self->_climateModuleId = self->_remoteControlCapabilities.climateControlCapabilities.firstObject.moduleInfo.moduleId;
+                
+        /// Get consent to control modules
+        SDLGetInteriorVehicleDataConsent *getInteriorVehicleDataConsent = [[SDLGetInteriorVehicleDataConsent alloc] initWithModuleType:SDLModuleTypeClimate moduleIds:@[self->_climateModuleId]];
+        [self->_sdlManager sendRequest:getInteriorVehicleDataConsent withResponseHandler:^(__kindof SDLRPCRequest * _Nullable request, __kindof SDLRPCResponse * _Nullable response, NSError * _Nullable error) {
+            if (!response.success) { return; }
+            
+            self->_consent = response.success;
+            
+            /// Initialize climate data and setup subscription
+            if (self->_consent) {
+                [self initializeClimateData];
+                [self subscribeVehicleData];
+                [self subscribeClimateControlData];
+            }
+        }];
+    }];
+
+    return self;
+}
+
+- (void)showClimateControl {
+    if (_climateModuleId == NULL && _consent) {
+        NSString *errorMessage = @"The climate module id was not set or consent was not given";
+        [AlertManager sendAlertWithManager:_sdlManager image:nil textField1:errorMessage textField2:nil];
+    }
+    
+    [_sdlManager.screenManager beginUpdates];
+    _sdlManager.screenManager.softButtonObjects = [self getButtons];
+    [_sdlManager.screenManager endUpdates];
```

Comment:
Generally, you always want to reference `self` unless you have a good reason not to (you're initing, copying, or you have a very good reason to skip the setter / getter)